Lib/_pyio.py

```diff
@@ -198,6 +198,10 @@ def open(file, mode="r", buffering=-1, encoding=None, errors=None,
         raise ValueError("binary mode doesn't take an errors argument")
     if binary and newline is not None:
         raise ValueError("binary mode doesn't take a newline argument")
+    if binary and buffering == 1:
+        import warnings
+        warnings.warn("line buffering is not supported in binary mode",
```

Comment:
It's not necessary obvious why the warning talks about line buffering, so I would be a bit more explicit: `"line buffering (buffering = 1) isn't supported in binary mode, file will be fully buffered."`

Reply:
OK, I've clarified the message as you suggest.